### PR TITLE
Friendships v1

### PR DESCRIPTION
**Pull request by AleksandarIvezic**
## In this Milestone we:
-Created a model with associations and all requested features for friendships (you can find all requirements in [Social Network](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/) )
-Created  unit and integrations tests

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
+    <%elsif current_user.requested?(@user)%>
+        <%= link_to "Accept", confirm_friendship_path(id: @user.id), method: :put, class: "btn-green"%>
+        <%= link_to "Decline", friendship_path(id: @user.id), method: :delete, data: {confirm: "Are you sure you want to decline friend request?"}, class: "btn-red"%>
+   <%elsif current_user.friend?(@user)%>
+    <%= link_to 'Remove Friend',  friendship_path(id: @user.id), method: :delete, data: {confirm: "Are you sure you want to remove friend?"}, class: "btn-red"%>
+   <%elsif !current_user.friend?(@user)%>
+    <%= link_to 'Invite to friendship',  friendships_path(user_id: @user.id), method: :post, action: :confirm, class: 'btn-green' %>
+    <% elsif !current_user.friend?(@user) %>
+    <%= link_to 'Remove Friendship',  friendship_path(id: @user.id), method: :delete, data: {confirm: "Are you sure you want to remove friend?"}, class: "btn-red"%>
+  <% end %>
```
Comment (SajjadAhmad14) [inline app/views/users/show.html.erb:16]: Please move this logic to app/helpers. [OPTIONAL]

File: app/views/users/_user.html.erb [hunk 2]
```diff
@@ -2,5 +2,17 @@
...
+        <span class = "bg-yellow">Current user</span>
+      <%elsif current_user.pending?(user)%>
+        <span class="bg-orange">Pending...</span>
+      <%elsif current_user.requested?(user)%>
+        <%= link_to "Accept", confirm_friendship_path(id: user.id), method: :put, class: "btn-green"%>
+        <%= link_to "Decline", friendship_path(id: user.id), method: :delete, data: {confirm: "Are you sure you want to decline friend request?"}, class: "btn-red"%>
+      <%elsif current_user.friend?(user)%>
+        <%= link_to 'Remove Friendship',  friendship_path(id: user.id), method: :delete, data: {confirm: "Are you sure you want to remove friend?"}, class: "btn-red"%>
+      <%elsif !current_user.friend?(user)%>
+        <%= link_to 'Invite to friendship',  friendships_path(user_id: user.id), method: :post, action: :confirm, class: 'btn-green' %>
```
Comment (SajjadAhmad14) [inline app/views/users/_user.html.erb:15]: Please move this logic to app/helpers. [OPTIONAL]

File: app/views/posts/_post.html.erb [hunk 3]
```diff
@@ -6,6 +6,9 @@
     <div class='post-liking'>
       <span class='post-date'> <%= post.created_at.strftime("%Y/%m/%d") %> </span>
       <%= like_or_dislike_btn(post) %>
+      <%if post.likes%>
+      <span>likes=<%= post.likes%></span>
+      <% end %>
```
Comment (SajjadAhmad14) [inline app/views/posts/_post.html.erb:11]: Please move this logic to app/helpers. [OPTIONAL]

File: app/views/friendships/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+<h2>Friend_requests</h2>
+<%current_user.friend_requests.each do |friend|%>
+<li><%= friend.name%></li>
+<%= link_to "Accept", confirm_friendship_path(id: friend.id), method: :put, class: "btn-green"%>
+<%= link_to "Decline", friendship_path(id: friend.id), method: :delete, data: {confirm: "Are you sure you want to decline friend request?"}, class: "btn-red"%>
+<% end %>
+<h3>Friends</h3>
+<%current_user.friends.each do |friend|%>
+<li><%= friend.name%></li>
+<% end %>
```
Comment (SajjadAhmad14) [inline app/views/friendships/index.html.erb:22]: Please move this logic to app/helpers. [OPTIONAL]

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, you have been doing a tremendous job so far :ok_hand:
## To Highlight 🎉
- [ ] You have mastered Git flow.
- [ ] PR title and summary are descriptive.
- [ ] All the specs are passing.
## To Improve 🙏
- [ ] Please add integration tests for the friendship feature.
- [ ] Please see inline comments for reference.
# Optional
- [ ] Views should be completely separate from app logic. You can move view-related logic to app/helpers.
### Happy Coding ☺️
## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Review by mke2111 — APPROVED** [review]
# 😄 🥇  STATUS: APPROVED  🥇 👏🏽 

## Hey Happy Coders, 👋🏽 
Congratulations, 🥂 . You got minimum requirements for approval. 👍🏽 

### **_Well Done_** 💯 

    1. The added tests cater to all methods. 👍🏽 

### You may now merge to the develop branch and remember to submit a project completion form.
### I wish you the very best in your upcoming projects.

**Cheers 🍹 and Happy coding 💻** 

<hr>

### Best regards
Slack 🆘  Roy Mukuye
Twitter 🐦  [Roy Mukuye](https://twitter.com/Roymkenya)
LinkedIn 🔗  [Roy Mukuye](https://www.linkedin.com/in/roy-mukuye/)

**Pull request merged**